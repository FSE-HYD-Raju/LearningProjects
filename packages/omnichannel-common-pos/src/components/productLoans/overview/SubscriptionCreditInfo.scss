@import "../../../variables";

.SubscriptionCreditInfo {
	&-title {
		font-size: 0.9rem;
		i {
			color: $main-blue;
			padding-left: 0.7rem;
		}
	}
	display: flex;
	align-items: center;
	justify-content: space-between;
	padding-bottom: 1rem;
}
