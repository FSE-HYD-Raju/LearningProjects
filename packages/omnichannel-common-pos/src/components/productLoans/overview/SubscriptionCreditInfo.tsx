import cssns from "../../../utils/cssnsConfig";
import OcCurrency from "../../ocComponents/OcCurrency";
import FormattedMessage from "../../../channelUtils/FormattedMessage";
import overviewProductLoanMessages from "./overviewProductLoan.messages";
import { ActiveProductLoan } from "../../../redux/models/productLoan/productLoan.types";
import CreditInfoModal from "./CreditInfoModal";
import { PureComponent } from "react";
const { React } = cssns("SubscriptionCreditInfo");

interface SubscriptionCreditInfoStateProps {
	loanDescription?: string;
	activeLoanDetails: ActiveProductLoan | undefined;
}
interface SubscriptionCreditInfoActionProps {
	actions: {
		openTopUpModal: () => void;
	};
}
interface SubscriptionCreditInfoProps extends SubscriptionCreditInfoStateProps, SubscriptionCreditInfoActionProps {}
interface SubscriptionCreditInfoState {
	showingCreditInfoModal: boolean;
}

class SubscriptionCreditInfo extends PureComponent<SubscriptionCreditInfoProps, SubscriptionCreditInfoState> {
	constructor(props: SubscriptionCreditInfoProps) {
		super(props);
		this.state = { showingCreditInfoModal: false };
	}
	showCreditInfoModal = () => {
		this.setState({ showingCreditInfoModal: true });
	};
	hideCreditInfoModal = () => {
		this.setState({ showingCreditInfoModal: false });
	};
	payBack = () => {
		this.hideCreditInfoModal();
		this.props.actions.openTopUpModal();
	};

	render() {
		const { activeLoanDetails, loanDescription } = this.props;
		if (!activeLoanDetails || !activeLoanDetails.totalAmountToPayBack.taxFreeAmount) {
			return null;
		}
		return (
			<div className="this">
				<div className="title">
					<FormattedMessage {...overviewProductLoanMessages.subscriptionCreditInfo} />
					<i id="credit-info-button" className="fa fa-info-circle" onClick={this.showCreditInfoModal} />
				</div>
				<div className="amount">
					<OcCurrency
						cost={-activeLoanDetails.totalAmountToPayBack.taxFreeAmount}
						currency={activeLoanDetails.totalAmountToPayBack.currency}
					/>
				</div>
				<CreditInfoModal
					showModal={this.state.showingCreditInfoModal}
					onPayBack={this.payBack}
					onClose={this.hideCreditInfoModal}
					loanDescription={loanDescription}
					originalLoanAmount={activeLoanDetails.loanedBalance}
					originalLoanFee={activeLoanDetails.loanFee}
					remainingLoanAmount={activeLoanDetails.remainingLoanAmount}
					loanAmountToPayBack={activeLoanDetails.loanAmountToPayBack}
					loanFeeToPayBack={activeLoanDetails.loanFeeToPayBack}
					totalRemainingCredit={activeLoanDetails.totalAmountToPayBack}
					dueDate={activeLoanDetails.dueDate}
				/>
			</div>
		);
	}
}

export {
	SubscriptionCreditInfoStateProps,
	SubscriptionCreditInfoProps,
	SubscriptionCreditInfoState,
	SubscriptionCreditInfoActionProps
};
export default SubscriptionCreditInfo;
