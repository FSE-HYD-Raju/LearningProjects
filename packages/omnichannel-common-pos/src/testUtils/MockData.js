import _ from "lodash";
import R from "ramda";

const firstDateThisWeek = () => {
	const date = new Date();
	while (date.getDay() !== 0) {
		date.setDate(date.getDate() - 1);
	}
	return date;
};

const people = [
	{
		firstName: "Kathrin",
		lastName: "K",
		gender: "female",
		language: "swe",
		countryOfBirth: "SE",
		birthDay: firstDateThisWeek(),
		emails: [{ email: "kathrin.k@qvantel.com" }],
		email: "kathrin.k@qvantel.com",
		mobileNumbers: [{ number: "2398457" }],
		mobileNumber: "2398457",
		fixedLineNumbers: [{ number: "2398457" }],
		fixedLineNumber: "2398457",
		password: "qwerasdf",
		passwordConfirmation: "qwerasdf"
	},
	{
		firstName: "John",
		lastName: "Smith",
		gender: "male",
		language: "fin",
		birthDay: firstDateThisWeek(),
		countryOfBirth: "United Kingdom",
		emails: [{ email: "john.smith@qvantel.com" }],
		email: "john.smith@qvantel.com",
		mobileNumbers: [{ number: "3498573" }],
		mobileNumber: "3498573",
		fixedLineNumbers: [{ number: "52497584" }],
		fixedLineNumber: "52497584"
	}
];

const basketItems = [
	{
		id: "7879c247-9e98-40ee-bb48-544eb366494a",
		type: "basketItems",
		attributes: {
			product: {
				name: "Movie Steaming Service"
			},
			totalPrices: [
				{
					type: "ONE_TIME",
					taxFreeAmount: 1.9,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: 14.9,
					taxRate: 0,
					currency: "EUR"
				}
			],
			childBasketItems: [],
			totalUpfrontPrice: 1.9,
			totalUpfrontPriceWithChildren: 1.9
		}
	},
	{
		id: "c0065ffc-77f4-4e44-b15c-fbd2f78d04ce",
		type: "basketItems",
		attributes: {
			product: {
				name: "XDSL"
			},
			totalPrices: [
				{
					type: "ONE_TIME",
					taxFreeAmount: 100.6,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: 30,
					taxRate: 0,
					currency: "EUR"
				}
			],
			childBasketItems: []
		}
	}
];

const deliveryItem = {
	id: "c0065ffc-77f4-4e44-b15c-fbd2f78d04ce",
	type: "basketItems",
	inputtedCharacteristics: {
		CH_Delivery_Contact_Info: "5c5dddf6-72be-4d0a-9711-6027f247af28"
	},
	product: {
		name: "Home delivery",
		id: "PO_Home_Delivery"
	}
};

export default class MockData {
	static getPerson(searchCriteria) {
		if (searchCriteria) {
			return _.find(people, searchCriteria);
		} else {
			return people[0];
		}
	}

	static getPeople() {
		return people;
	}

	static getPostalAddress() {
		return {
			street: "1 Test St",
			coAddress: "Best Ltd.",
			postalCode: "90210",
			city: "Test City",
			country: "FI"
		};
	}

	static getUser() {
		return {
			"id": "bf588945-ce95-490a-a774-8ef91661a146",
			"type": "persons",
			"attributes": {
				"lastName": "Testeri",
				"addresses": null,
				"gender": "male",
				"fixedLineNumbers": [],
				"formattedName": "Teppo Testeri",
				"roles": null,
				"billingAccountId": null,
				"language": null,
				"emails": [
					{
						"role": "PRIMARY",
						"id": "2ab79d7b-862d-49df-8afc-e6104a914919",
						"email": "tepsteri@tepinposti.fi"
					}
				],
				"honorificSuffix": null,
				"countryOfBirth": "FI",
				"personRelationships": [],
				"birthDay": null,
				"placeOfBirth": null,
				"characteristics": null,
				"mobileNumbers": [
					{
						"role": "PRIMARY",
						"id": "a843f46d-f729-40dd-a7ce-240a64bc008e",
						"number": "1234576890"
					}
				],
				"honorificPrefix": null,
				"avatar": "/img/icon-user-male.svg",
				"privacySettings": {
					"privacy1": false,
					"privacy2": false
				},
				"postalAddresses": [
					{
						"role": "PRIMARY",
						"id": "5e30221d-ad19-4043-8088-f594f5fdf77f",
						"street": "Piippukatu 11 A 1",
						"coAddress": null,
						"postalCode": "40100",
						"city": "Roma",
						"country": "FI",
						"county": null,
						"description": null,
						"postOfficeBox": null,
						"stateOrProvince": null,
						"addressRegisterId": null,
						"apartment": null,
						"building": null
					},
					{
						"role": "DELIVERY",
						"id": "5c5dddf6-72be-4d0a-9711-6027f247af28",
						"street": "Piippukatu 11 A 1",
						"coAddress": "Piippukatu 11 A 1",
						"postalCode": "40100",
						"city": "Roma",
						"country": "FI",
						"county": null,
						"description": null,
						"postOfficeBox": null,
						"stateOrProvince": "",
						"addressRegisterId": null,
						"apartment": null,
						"building": null
					}
				],
				"identifications": [],
				"firstName": "Teppo",
				"customerAccountId": null,
				"nationality": null,
				"additionalName": null,
				"maritalStatus": null
			}
		}
	}

	static getBasket(withItems: boolean, withDeliveryItems: boolean) {
		const basket = {
			activeBasket: {
				attributes: {
					totalPrices: [],
					totalUpfrontPrice: 0
				}
			}
		};

		if (withItems) {
			basket.basketItems = basketItems;

			const totalPriceOneTime = R.reduce(
				R.add,
				0,
				basketItems.map(bi => {
					return bi.attributes.totalPrices[0].taxFreeAmount;
				})
			);
			const totalPriceMonthly = R.reduce(
				R.add,
				0,
				basketItems.map(bi => {
					return bi.attributes.totalPrices[0].taxFreeAmount;
				})
			);

			basket.activeBasket.attributes.totalPrices = [
				{
					type: "ONE_TIME",
					taxFreeAmount: totalPriceOneTime,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: totalPriceMonthly,
					taxRate: 0,
					currency: "EUR"
				}
			];

			if (withDeliveryItems) {
				basket.basketItems[0].attributes.childBasketItems.push(
					deliveryItem
				);
			}
		}

		return basket;
	}
}
