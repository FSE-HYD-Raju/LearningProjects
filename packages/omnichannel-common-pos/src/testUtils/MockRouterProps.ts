import {
	RouteComponentProps,

} from "react-router";

import {
	Href,
	UnregisterCallback,
	TransitionPromptHook,
	LocationState,
	Path,
	LocationDescriptorObject,
	LocationListener
} from "history";

const mockLocation: RouteComponentProps<any>["location"] = {
		hash: "",
		key: "",
		pathname: "",
		search: "",
		state: {}
};

const mockHistory: RouteComponentProps<any>["history"] = {
	length: 2,
	action: "POP",
	location: mockLocation,
	push: (path: Path | LocationDescriptorObject, state?: LocationState): void => {},
	replace: (path: Path | LocationDescriptorObject, state?: LocationState): void => {},
	go: (n: number): void => {},
	goBack: (): void => {},
	goForward: (): void => {},
	block: (prompt?: boolean | string | TransitionPromptHook): UnregisterCallback => { return () => {}; },
	listen: (listener: LocationListener): UnregisterCallback => {return () => {}; },
	createHref: (location: LocationDescriptorObject): Href => { return ""; },
};

const mockRouterProps: RouteComponentProps<any> = {
		match: {
			isExact: true,
			params: {},
			path: "",
			url: ""
		},
		location: mockLocation,
		history: mockHistory,
		staticContext: {}
};

export {
	mockRouterProps,
	mockLocation,
	mockHistory,
	RouteComponentProps
};
