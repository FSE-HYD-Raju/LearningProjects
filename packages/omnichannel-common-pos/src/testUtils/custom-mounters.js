import React from "react";
import { IntlProvider, intlShape } from "react-intl";
import { mount, shallow } from "enzyme";
import PropTypes from "prop-types";
import R from "ramda";
import TestUtils from "./TestUtils";

const MAX_DIVE_DEEP_LEVEL = 16;
const intlProvider = new IntlProvider({ locale: "en" }, {});
const { intl } = intlProvider.getChildContext();

const baseContext = {
	flux: {
		persist: () => {}
	},
	store: TestUtils.mockReduxStore(),
	intl,
	location: {},
	only: {},
	skip: {}
};

const nodeWithIntlProp = (node, mockedRouterOptions) => {
	if (mockedRouterOptions) {
		mockedRouterOptions.props().history.push = jest.fn();
	}
	return React.cloneElement(node, {
		intl,
		...(mockedRouterOptions ? mockedRouterOptions.props() : {})
	});
};

function getOptionsForShallowMount(options, debug, mockedRouterContextOptions) {
	const mountOptions = R.merge(options ? options : {}, {
		disableLifecycleMethods: !(
			options && options.disableLifecycleMethods === false
		),
		context: R.mergeAll([
			debug
				? {
						...baseContext
					}
				: baseContext,
			R.omit("intl", options && options.context ? options.context : {}),
			mockedRouterContextOptions.context
		]),
		childContextTypes: R.merge(
			{
				intl: intlShape,
				store: PropTypes.object
			},
			mockedRouterContextOptions.childContextTypes
		)
	});
	TestUtils.mockReduxStoreToFlux(mountOptions.context);
	return mountOptions;
}
function deepDive(wrapper) {
	let diveAttemptsCount = 0;
	while (
		wrapper.hostNodes().length === 0 &&
		diveAttemptsCount < MAX_DIVE_DEEP_LEVEL
	) {
		wrapper = wrapper.dive();
		diveAttemptsCount += 1;
	}
	return wrapper;
}
export function shallowWithContextAndRouterProps(node, options, debug) {
	return shallowWithContext(node, options, debug, true);
}
export function shallowWithContextAndLifecycle(node, options, debug) {
	return shallowWithContext(
		node,
		R.merge(options || {}, { disableLifecycleMethods: false }),
		debug,
		true
	);
}
export function deepShallowWithContextAndLifecycle(node, options, debug) {
	return deepDive(shallowWithContextAndLifecycle(node, options, debug));
}
export function deepShallowWithContext(node, options, debug) {
	return deepDive(shallowWithContext(node, options, debug, true));
}
export function shallowWithContext(node, options, debug, injectRouterProps) {
	const mockedRouterOptions = TestUtils.getMockedRouterOptions();
	const mockedRouterContextOptions = mockedRouterOptions.get();
	return shallow(
		nodeWithIntlProp(node, injectRouterProps ? mockedRouterOptions : null),
		getOptionsForShallowMount(options, debug, mockedRouterContextOptions)
	);
}

export function mountWithContextAndRouterProps(node, options, debug) {
	return mountWithContext(node, options, debug, true);
}
export function mountWithContext(node, options, debug, injectRouterProps) {
	const mockedRouterOptions = TestUtils.getMockedRouterOptions();
	const mockedRouterContextOptions = mockedRouterOptions.get();
	const mountOptions = R.merge(options ? options : {}, {
		context: R.mergeAll([
			debug
				? {
						...baseContext
					}
				: baseContext,
			R.omit("intl", options && options.context ? options.context : {}),
			mockedRouterContextOptions.context
		]),
		childContextTypes: R.merge(
			{
				flux: PropTypes.object,
				store: PropTypes.object,
				intl: intlShape,
				location: PropTypes.object,
				only: PropTypes.object,
				skip: PropTypes.object
			},
			mockedRouterContextOptions.childContextTypes
		)
	});
	TestUtils.mockReduxStoreToFlux(mountOptions.context);
	return mount(
		nodeWithIntlProp(node, injectRouterProps ? mockedRouterOptions : null),
		mountOptions
	);
}

const attachmentElementId = "__OC_COMPONENT_TEST_ATTACHMENT_ELEMENT__";

export function attachWithContext(node, options) {
	const attachmentElement = document.createElement("div");
	attachmentElement.setAttribute("id", attachmentElementId);
	document.body.appendChild(attachmentElement);

	const wrapper = mountWithContext(
		node,
		R.merge(options, { attachTo: attachmentElement })
	);
	const doppelWrapper = hijackDetacher(wrapper);

	return doppelWrapper;
}

function hijackDetacher(wrapper) {
	const doppel = wrapper;
	doppel.__detach = wrapper.detach;

	doppel.detach = () => {
		removeAttachmentElements();
		doppel.__detach.apply(wrapper);
	};

	return doppel;
}

function removeAttachmentElements() {
	document.body.querySelectorAll("#" + attachmentElementId).forEach(elem => {
		document.body.removeChild(elem);
	});
}

export function detach(wrapper) {
	removeAttachmentElements();
	wrapper.detach();
}
