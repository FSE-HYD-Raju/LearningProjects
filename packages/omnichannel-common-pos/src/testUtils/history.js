import isClient from "../utils/isClient";
import { createBrowserHistory, createMemoryHistory } from "history";
import { useRouterHistory } from "react-router";

const history = isClient
	? useRouterHistory(createBrowserHistory)({ queryKey: false })
	: createMemoryHistory();

export default history;
