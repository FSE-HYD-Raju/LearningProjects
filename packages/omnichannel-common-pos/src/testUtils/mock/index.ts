import { default as productOffering } from "./ProductOfferingMock";
import { ProductMock as product } from "./ProductMock";
import { AgreementMock as agreement } from "./AgreementMock";
import { ChargingBalancesMock as chargingBalances } from "./ChargingBalancesMock";
import { default as postalAddress } from "./PostalAddressMock";
import { default as identification } from "./IdentificationMock";
import { default as user } from "./UserMock";
import { default as order } from "./OrderMock";
import { default as basket } from "./BasketMock";
export { productOffering, postalAddress, identification, user, order, basket, product, agreement, chargingBalances };
