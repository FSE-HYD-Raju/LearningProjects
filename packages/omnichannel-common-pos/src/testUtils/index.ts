import ReactWidgetsTestUtils from "./ReactWidgetsTestUtils";
import * as MockDataMaker from "./mock";
const {
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach
} = require("./custom-mounters");
const TestUtils = require("./TestUtils");
const MockData = require("./MockData");
const SimpleDataMock = require("./mock/simpleDataMock");

export {
	MockData,
	ReactWidgetsTestUtils,
	TestUtils,
	SimpleDataMock,
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach,
	MockDataMaker
};

export * from "./MockRouterProps";

export default TestUtils;
