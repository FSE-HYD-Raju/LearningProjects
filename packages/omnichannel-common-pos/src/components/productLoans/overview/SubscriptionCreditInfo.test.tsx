import * as React from "react";
import SubscriptionCreditInfo, { SubscriptionCreditInfoProps } from "./SubscriptionCreditInfo";
import { shallowWithContext, mountWithContext } from "../../../testUtils";
import {
	ProductLifecycleStatus,
	SimplePrice
} from "../../../redux/types";

const getPrice = (cost: number): SimplePrice => ({ taxFreeAmount: cost, currency: "EUR" });

describe("SubscriptionCreditInfo", () => {
	const props: SubscriptionCreditInfoProps = {
		loanDescription: "test description",
		activeLoanDetails: {
			loanFee: getPrice(1),
			loanedBalance: getPrice(2),
			remainingLoanAmount: getPrice(3),
			loanAmountToPayBack: getPrice(4),
			loanFeeToPayBack: getPrice(5),
			totalAmountToPayBack: getPrice(6),
			dueDate: new Date("October 30, 2018 23:15:30 GMT+02:00")
		},
		actions: {
			openTopUpModal: jest.fn()
		}
	};

	it("succeeds at shallow mount with minimal props", () => {
		const wrapper = shallowWithContext(<SubscriptionCreditInfo {...props} />);
		expect(wrapper).toMatchSnapshot();
	});

	it("should render null when no active loan", () => {
		const wrapper = shallowWithContext(<SubscriptionCreditInfo {...{ ...props, activeLoanDetails: undefined }} />);
		expect(wrapper.getElement()).toBeNull();
	});
	it("should show modal when clicked on info circle", () => {
		const wrapper = mountWithContext(<SubscriptionCreditInfo {...props} />);
		wrapper.find("#credit-info-button").simulate("click");
		wrapper.update();
		expect(wrapper.find("CreditInfoModal").props().showModal).toBeTruthy();
	});
	it("should hide modal when clicked modal close button", () => {
		const wrapper = mountWithContext(<SubscriptionCreditInfo {...props} />);
		wrapper.find("#credit-info-button").simulate("click");
		wrapper.update();
		wrapper.find(".modal-header button").simulate("click");
		wrapper.update();
		expect(wrapper.find("CreditInfoModal").props().showModal).toBeFalsy();
	});
});
