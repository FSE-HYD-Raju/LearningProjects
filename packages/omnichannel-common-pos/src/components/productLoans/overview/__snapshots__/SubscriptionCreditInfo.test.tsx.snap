// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscriptionCreditInfo succeeds at shallow mount with minimal props 1`] = `
<div
  className="SubscriptionCreditInfo"
>
  <div
    className="SubscriptionCreditInfo-title"
  >
    <FormattedMessageWrapper
      defaultMessage="Credit"
      description="Subscription overview (balances section) credit info title"
      id="subscription-overview-credit-info-title"
      wrapperType="wrapper"
    />
    <i
      className="fa fa-info-circle"
      id="credit-info-button"
      onClick={[Function]}
    />
  </div>
  <div
    className="SubscriptionCreditInfo-amount"
  >
    <OcCurrency
      cost={-6}
      currency="EUR"
    />
  </div>
  <CreditInfoModal
    dueDate={2018-10-30T21:15:30.000Z}
    loanAmountToPayBack={
      Object {
        "currency": "EUR",
        "taxFreeAmount": 4,
      }
    }
    loanDescription="test description"
    loanFeeToPayBack={
      Object {
        "currency": "EUR",
        "taxFreeAmount": 5,
      }
    }
    onClose={[Function]}
    onPayBack={[Function]}
    originalLoanAmount={
      Object {
        "currency": "EUR",
        "taxFreeAmount": 2,
      }
    }
    originalLoanFee={
      Object {
        "currency": "EUR",
        "taxFreeAmount": 1,
      }
    }
    remainingLoanAmount={
      Object {
        "currency": "EUR",
        "taxFreeAmount": 3,
      }
    }
    showModal={false}
    totalRemainingCredit={
      Object {
        "currency": "EUR",
        "taxFreeAmount": 6,
      }
    }
  />
</div>
`;
